Sources/RSocketCore/DuplexConnection.swift

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Foundation
+import NIO
+
+
+/// Represents a connection with input/output that the protocol uses.
+public protocol DuplexConnection: Closeable {
+
+    ///
+    /// - Returns the assigned `ByteBufAllocator`.
+    ///
+    var alloc: ByteBufferAllocator { get }
```

Comment:
Could we rename this to `allocator`?
```suggestion
    var allocator: ByteBufferAllocator { get }
```